### PR TITLE
[ISSUE #12017] Split console authentication

### PR DESCRIPTION
**Pull request by RickonZhang0929**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

For #12017

Split console authentication.

## Brief changelog

* Add ApiType annotations

* Update configuration file to add fields

* Update console authentication status fetch

* Update `application.properties` - Reuse `nacos.core.auth.enabled` and add `nacos.core.auth.console.enabled`

* Update `AuthConfigs` - Modify `isAuthEnabled()` to manage plugin initialization

* Update `AbstractProtocolAuthService` - Implement `isAuthEnabled()` method for `Secured` annotation and configuration-based authentication

* Update `RemoteRequestAuthFilter` - Use `authConfigs.isAuthEnabled()` for initialization and `protocolAuthService.authEnabled(secured)` for authentication checks

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by KomachiSion — COMMENTED** [review]
It seems no relative with auth？

**Comment by RickonZhang0929** [comment]
> It seems no relative with auth？

This one is for testing commits, I'll fix this change

File: auth/src/main/java/com/alibaba/nacos/auth/AbstractProtocolAuthService.java [hunk 1]
```diff
@@ -57,6 +58,17 @@ public boolean enableAuth(Secured secured) {
         return false;
     }
     
+    @Override
+    public boolean authEnabled(Secured secured) {
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/AbstractProtocolAuthService.java:62]: 和上述的enableAuth区别是什么？

是不是应该放到NacosAuthPluginService的enableAuth里？

File: auth/src/main/java/com/alibaba/nacos/auth/ProtocolAuthService.java [hunk 2]
```diff
@@ -49,6 +49,14 @@ public interface ProtocolAuthService<R> {
      */
     boolean enableAuth(Secured secured);
     
+    /**
+     * Determine whether to enable authentication according to secured information.
+     *
+     * @param secured secured information
+     * @return {@code true} if auth is open, otherwise {@code false}
+     */
+    boolean authEnabled(Secured secured);
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/ProtocolAuthService.java:58]: 和上一个enable auth看起来重复，让插件开发和摸不着头脑

File: auth/src/main/java/com/alibaba/nacos/auth/enums/ApiType.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * The type of API.
+ *
+ * @author zhangyukun
+ */
+public enum ApiType {
+    /**
+     * console API.
+     */
+    ADMIN_API("ADMIN_API"),
```
Comment (KomachiSion) [inline auth/src/main/java/com/alibaba/nacos/auth/enums/ApiType.java:29]: 我想了一下， 这里可能改叫CONSOLE_API比较好

Comment (KomachiSion) [reply auth/src/main/java/com/alibaba/nacos/auth/enums/ApiType.java:29]: 因为之后可能会单独出maintainer sdk访问admin api，同时拆分部署后console访问server的api可能才叫ADMIN API

所以我想目前的叫CONSOLE_API

当然之后统一改也可以

File: core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java [hunk 4]
```diff
@@ -115,6 +115,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 }
                 
                 Secured secured = method.getAnnotation(Secured.class);
+                if (!protocolAuthService.authEnabled(secured)) {
```
Comment (KomachiSion) [inline core/src/main/java/com/alibaba/nacos/core/auth/AuthFilter.java:118]: 和前面描述的一样，不再赘述

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**